services_delegate.go

```diff
@@ -117,9 +124,18 @@ func (d *servicesDelegate) GetBroadcasts(overhead, limit int) [][]byte {
 	log.Debugf("Sending broadcast %d msgs %d 1st length",
 		len(broadcast), len(broadcast[0]),
 	)
-	if len(leftover) > 0 {
-		log.Debugf("Leaving %d messages unsent", len(leftover))
-	}
+
+	// Unfortunately Memberlist does not provide a callback after broadcasts were
+	// accepted so we have no direct way to return these to the pool. However, it
+	// immediately copies what we return into a new buffer. So, it's not perfectly,
+	// but is reasonably safe to wait awhile and then re-add our buffer to the
+	// ffjson pool.
+	go func(broadcast [][]byte) {
+		time.Sleep(25 * time.Millisecond) // Lots of safety margin in this number
+		for i := 0; i < len(broadcast); i++ {
+			ffjson.Pool(broadcast[i])
+		}
+	}(broadcast)
```

Comment:
Broadcast is a slice that gets returned. The slice, being a pointer, could be manipulated to point to something different by the time the goroutine does the work.